src/Samples/DesktopServer/EchoService.cs

```diff
@@ -1,25 +1,76 @@
-﻿using System.ServiceModel;
+using Microsoft.Extensions.Logging;
 
-namespace DesktopServer
+#if NETFRAMEWORK
+using System.ServiceModel;
+#else
+using CoreWCF;
+#endif
+
+using Contract;
+
+namespace ServerLogic
 {
-    public class EchoService : Contract.IEchoService
+    [ServiceBehavior(IncludeExceptionDetailInFaults = true)]
+    public class EchoService : IEchoService
     {
+        private FaultException CreateFault<T>(T detail, string reason, string code)
+        {
+#if NETFRAMEWORK
+            return new System.ServiceModel.FaultException<T>(detail, new FaultReason(reason), new FaultCode(code));
```

Comment:
Where is there conditional compilation here? This file is in the DesktopServer folder so will only be used with .NET Framework.

Reply:
This is a copy of the file in src/Samples/NetCoreServer/EchoService.cs, 
Thought in creating a Shared project to avoid duplication

Reply:
I personaly think that sample's code should allow users to quickly get how it is expected to work and be configured, without any complexity. Thus factorizing this code lead to user not being able to copy paste and increase the time required to figure out the required steps to get started.
Please have a look to the below repos to see how they are duplicating code to ease user understanding:
https://github.com/IdentityServer/IdentityServer4/tree/main/samples
https://github.com/dotnet/roslyn-sdk/tree/main/samples/CSharp/SourceGenerators/SourceGeneratorSamples

Reply:
The reasoning behind using #if NETFRAMEWORK is to indicate that System.ServiceModel should be used with dot net framework otherwise CoreWCF in case a FaultException needs to be thrown.

Thus, people migrating or trying new functionality won't lose time figuring out why Faults might not work if this is not taken into account.

I did not use a Share Project because indeed thought might not be necessary

Reply:
Should be fixed by https://github.com/CoreWCF/CoreWCF/pull/488

Reply:
NetCoreServer still fails if System.ServiceModel.FaultException<EchoFault> is thrown.
It works when using CoreWCF<EchoFault>

Reply:
Can you confirm you are trying to throw a SSM Fault.Exception from your CoreWCF codebase ?
If this is the case the related issue is https://github.com/CoreWCF/CoreWCF/issues/103.
However i do not understand why this really specific use case is in a sample.

Reply:
Yes, NetCoreServer uses CoreWCF and #103 describes the issue.

Given that the general MS guidance is to use other technologies, it should be expected that most code should try to use System.ServiceModel types instead of CoreWCF as the most common case will be migration.

Examples should try to guide CoreWCF adopters to cover the most common scenarios, such as returning Faults or using Dependency injection. A complementary alternative is to provide documentation and/or tutorials.

Although the information can be found in issues, is common that the solution/workaround is not clear and takes important time to find the right way to apply.

Reply:
btw, just noticed #488 hasn't been completed yet, so my latest test was with just with the lastest main.